include/miral/miral/wayland_extensions.h

```diff
@@ -53,6 +61,8 @@ class WaylandExtensions
     /// Initialize "enabled by default" to a custom set of extensions (colon
     /// separated list).
     /// \note This can only be a subset of supported_extensions()
+    /// \deprecated A better option is to use the default constructor, enable()
+    /// and disable()
```

Comment:
Having thought about it, I don't think we should deprecate this. It covers scenarios that are not addressed otherwise: where the compositor needs to support a specific set of extensions (i.e. not additions & deletions from default.)

Reply:
In that case, perhaps we should add another constructor that takes a `std::vector<std::string>` so we can deprecate the colon list being used anywhere but command line options.